### PR TITLE
Random fixes for issues reported by clang-check

### PR DESCRIPTION
**Pull request by yamt**
## Summary
Random fixes for issues reported by clang-check
## Impact
None of these seems critical
## Testing
Compile tested only

### CODE REVIEW

File: arch/sim/src/sim/up_exit.c [hunk 1]
```diff
@@ -65,9 +65,9 @@
 
 void up_exit(int status)
 {
-  FAR struct tcb_s *tcb = this_task();
+  FAR struct tcb_s *tcb;
```
Comment (v01d) [inline arch/sim/src/sim/up_exit.c:68]: what does this change accomplish? I can only see that it introduces a second call to `this_task()`.

File: net/tcp/tcp_timer.c [hunk 2]
```diff
@@ -182,7 +182,7 @@ void tcp_timer(FAR struct net_driver_s *dev, FAR struct tcp_conn_s *conn,
 
               /* Notify upper layers about the timeout */
 
-              result = tcp_callback(dev, conn, TCP_TIMEDOUT);
+              tcp_callback(dev, conn, TCP_TIMEDOUT);
```
Comment (v01d) [inline net/tcp/tcp_timer.c:185]: is this correct? what did clang-check say?

Comment (v01d) [reply net/tcp/tcp_timer.c:185]: ah, I see the commit message. the fact that result is never read probably indicates that it should be read. not assigning the result is not the correct solution if so.

File: arch/sim/src/sim/up_exit.c (hunk 1 above)
Comment (yamt) [reply arch/sim/src/sim/up_exit.c:68]: the commit messages have the snippets of the clang output.

this change is intended to appease the warning when sinfo() expands to empty.

File: drivers/usbhost/hid_parser.c [hunk 3]
```diff
@@ -274,8 +274,8 @@ int hid_parsereport(FAR const uint8_t *report, int rptlen,
             {
               collectionpath->usage.usage = usage[0];
 
-              for (i = 0; i < nusage; i++)
-                usage[i] = usage[i + 1];
+              for (i = 1; i < nusage; i++)
```
Comment (v01d) [inline drivers/usbhost/hid_parser.c:286]: I understand the code does the same thing as before, only differently. I don't understand by clang-check considers the other version problematic and this one does not.

File: net/tcp/tcp_timer.c (hunk 2 above)
Comment (yamt) [reply net/tcp/tcp_timer.c:185]: see the commit message for the clang output.
i read the code and i agreed with clang about this.

File: arch/sim/src/sim/up_exit.c (hunk 1 above)
Comment (v01d) [reply arch/sim/src/sim/up_exit.c:68]: Ok, I understand.

File: drivers/usbhost/hid_parser.c (hunk 3 above)
Comment (yamt) [reply drivers/usbhost/hid_parser.c:286]: the original code reads usage[nusage+1]. the new one doesn't.

File: net/tcp/tcp_timer.c (hunk 2 above)
Comment (v01d) [reply net/tcp/tcp_timer.c:185]: ok, tcp_timer does not return errors so I guess it is OK

File: drivers/usbhost/hid_parser.c (hunk 3 above)
Comment (v01d) [reply drivers/usbhost/hid_parser.c:286]: got it

**Pull request merged**